### PR TITLE
FDG-4340: support for light query engine

### PR DESCRIPTION
**Pull request by dileepk7**

### CODE REVIEW

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java [hunk 1]
```diff
@@ -901,6 +912,18 @@ private boolean connectUsingArgs(BeelineParser beelineParser, CommandLine cl) {
...
+
+      String sessionHookClassName = getOpts().getHiveConfVariables()
+              .getOrDefault(BEELINE_SESSION_HOOK_CLASS, BEELINE_DEFAULT_SESSION_HOOK_CLASS);
+      try {
+        BeelineSessionHook sessionHook =
+                (BeelineSessionHook) Class.forName(sessionHookClassName).newInstance();
+        setSessionHook(sessionHook);
+        sessionHook.executePreHook(getOpts().getHiveConfVariables(), user, url);
+      } catch (Throwable e) {
+        throw new RuntimeException(e.getMessage());
```
Comment (kartikaditya) [inline beeline/src/java/org/apache/hive/beeline/BeeLine.java]: throw new RuntimeException(e);

File: common/src/java/org/apache/hadoop/hive/conf/HiveConf.java [hunk 2]
```diff
@@ -4432,7 +4432,15 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
...
         "org.apache.hadoop.hive.ql.propertymodifier.NoEnrichment",
-        "Fully qualified name of class to be used to enrich job name");
+        "Fully qualified name of class to be used to enrich job name"),
+
+    /**
+     * Properties to support light query engine
+     */
+    LIGHT_QUERY_ENGINE_ENABLED("light.query.engine.enabled", false,
+                                       "Checks if light query engine is enabled"),
+    LIGHT_QUERY_ENGINE_QUEUE("light.query.engine.queue", "default_queue",
```
Comment (kartikaditya) [inline common/src/java/org/apache/hadoop/hive/conf/HiveConf.java]: We should not have a default value

Comment (dileepk7) [reply common/src/java/org/apache/hadoop/hive/conf/HiveConf.java]: Discussed with Kartik, It is fine.

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java (hunk 1 above)
Comment (dileepk7) [reply beeline/src/java/org/apache/hive/beeline/BeeLine.java]: done

**Review by kartikaditya — APPROVED** [review]
LGTM

**Review by jithendhir92 — APPROVED** [review]
LGTM

**Review by kartikaditya — APPROVED** [review]
LGTM

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java [hunk 3]
```diff
@@ -162,6 +163,9 @@
   public static final String BEELINE_DEFAULT_JDBC_DRIVER = "org.apache.hive.jdbc.HiveDriver";
   public static final String DEFAULT_DATABASE_NAME = "default";
 
+  private static final String BEELINE_SESSION_HOOK_CLASS = "beeline.session.hook.class";
+  private static final String BEELINE_DEFAULT_SESSION_HOOK_CLASS = "org.apache.hive.beeline.DefaultBeelineSessionHook";
```
Comment (mittalnanu) [inline beeline/src/java/org/apache/hive/beeline/BeeLine.java]: Change the variable name to BEELINE_SESSION_HOOK_CLASS_VARVALUE_DEFAULT

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java [hunk 4]
```diff
@@ -162,6 +163,9 @@
   public static final String BEELINE_DEFAULT_JDBC_DRIVER = "org.apache.hive.jdbc.HiveDriver";
   public static final String DEFAULT_DATABASE_NAME = "default";
 
+  private static final String BEELINE_SESSION_HOOK_CLASS = "beeline.session.hook.class";
```
Comment (mittalnanu) [inline beeline/src/java/org/apache/hive/beeline/BeeLine.java]: change to 
BEELINE_SESSION_HOOK_CLASS_VARNAME

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java [hunk 5]
```diff
@@ -901,6 +912,18 @@ private boolean connectUsingArgs(BeelineParser beelineParser, CommandLine cl) {
...
+
+      String sessionHookClassName = getOpts().getHiveConfVariables()
+              .getOrDefault(BEELINE_SESSION_HOOK_CLASS, BEELINE_DEFAULT_SESSION_HOOK_CLASS);
+      try {
+        BeelineSessionHook sessionHook =
+                (BeelineSessionHook) Class.forName(sessionHookClassName).newInstance();
+        setSessionHook(sessionHook);
+        sessionHook.executePreHook(getOpts().getHiveConfVariables(), user, url);
+      } catch (Throwable e) {
+        throw new RuntimeException(e);
```
Comment (mittalnanu) [inline beeline/src/java/org/apache/hive/beeline/BeeLine.java:924]: can we log the message in error

File: common/src/java/org/apache/hadoop/hive/conf/HiveConf.java [hunk 6]
```diff
@@ -4432,7 +4432,13 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "Fully Qualified class name for concerete impl to be used to enforce queues"),
     JOBNAME_ENRICHER_CLASS("jobname.enricher.class",
         "org.apache.hadoop.hive.ql.propertymodifier.NoEnrichment",
-        "Fully qualified name of class to be used to enrich job name");
+        "Fully qualified name of class to be used to enrich job name"),
+
```
Comment (mittalnanu) [inline common/src/java/org/apache/hadoop/hive/conf/HiveConf.java:4436]: Do we need hive server deployment? If yes can be avoided?

**Review by mittalnanu — CHANGES_REQUESTED** [review]

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java (hunk 3 above)
Comment (dileepk7) [reply beeline/src/java/org/apache/hive/beeline/BeeLine.java]: done

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java (hunk 4 above)
Comment (dileepk7) [reply beeline/src/java/org/apache/hive/beeline/BeeLine.java]: done.

File: beeline/src/java/org/apache/hive/beeline/BeeLine.java (hunk 5 above)
Comment (dileepk7) [reply beeline/src/java/org/apache/hive/beeline/BeeLine.java:924]: Not needed.

File: common/src/java/org/apache/hadoop/hive/conf/HiveConf.java (hunk 6 above)
Comment (dileepk7) [reply common/src/java/org/apache/hadoop/hive/conf/HiveConf.java:4436]: Discussed. Currently we will go with this.

**Review by mittalnanu — APPROVED** [review]

**Pull request merged**